### PR TITLE
Prep lightning-persister for export in the C bindings

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

File: lightning-persister/src/lib.rs [hunk 1]
```diff
@@ -53,12 +52,12 @@ impl<Signer: Sign> DiskWriteable for ChannelMonitor<Signer> {
...
-      T: BroadcasterInterface,
-      K: KeysInterface<Signer=Signer>,
-      F: FeeEstimator,
-      L: Logger,
+impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> DiskWriteable for ChannelManager<Signer, M, T, K, F, L>
+where M::Target: chain::Watch<Signer>,
+      T::Target: BroadcasterInterface,
+      K::Target: KeysInterface<Signer=Signer>,
+      F::Target: FeeEstimator,
+      L::Target: Logger
```
Comment (jkczyz) [inline lightning-persister/src/lib.rs:60]: nit: could you format according to these guidelines?

https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/items.md#where-clauses

i.e., move the first item in the where clause to its own line, ident only with tabs, and use a trailing comma for all items (including the last one).

I know we haven't been very consistent, but this feels like a good convention to follow given it sticks to using only tabs and seems to make editors happy.

File: lightning-persister/src/lib.rs [hunk 2]
```diff
@@ -87,16 +86,15 @@ impl FilesystemPersister {
...
-	      M: chain::Watch<Signer>,
-	      T: BroadcasterInterface,
-	      K: KeysInterface<Signer=Signer>,
-	      F: FeeEstimator,
-	      L: Logger
+	where M::Target: chain::Watch<Signer>,
+	      T::Target: BroadcasterInterface,
+	      K::Target: KeysInterface<Signer=Signer>,
+	      F::Target: FeeEstimator,
+	      L::Target: Logger
```
Comment (jkczyz) [inline lightning-persister/src/lib.rs:97]: Likewise

**Review by jkczyz — APPROVED** [review]

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Actually, @valentinewallace is there any reason for `read_channelmonitors` to be a `HashMap` to begin with? Its guaranteed to be unique because its file-based, and the `OutPoint` key is available in the `ChannelMonitor` anyway, plus when we go to read the `ChannelManager` they need to be in a HashMap, but without the BlockHash in the pair, so we have to move/convert it anyway?

**Comment by TheBlueMatt** [comment]
Completely dropped the second commit and replaced it with one which just exports a `Vec` instead, which the bindings already handle and which is just as useful for users.

**Review by valentinewallace — APPROVED** [review]

File: lightning-persister/src/lib.rs [hunk 3]
```diff
@@ -52,11 +52,12 @@ impl<Signer: Sign> DiskWriteable for ChannelMonitor<Signer> {
...
-      T::Target: BroadcasterInterface,
-      K::Target: KeysInterface<Signer=Signer>,
-      F::Target: FeeEstimator,
-      L::Target: Logger
+where
+	M::Target: chain::Watch<Signer>,
+	T::Target: BroadcasterInterface,
+	K::Target: KeysInterface<Signer=Signer>,
+	F::Target: FeeEstimator,
+	L::Target: Logger
```
Comment (jkczyz) [inline lightning-persister/src/lib.rs:60]: Add trailing comma on last item here and below.

**Review by jkczyz — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Only trivial style changes, will take after CI:
```
$ git diff-tree -U2 651f69b8 3fdf252b
diff --git a/lightning-persister/src/lib.rs b/lightning-persister/src/lib.rs
index c930de26..f6fbc635 100644
--- a/lightning-persister/src/lib.rs
+++ b/lightning-persister/src/lib.rs
@@ -58,5 +58,5 @@ where
 	K::Target: KeysInterface<Signer=Signer>,
 	F::Target: FeeEstimator,
-	L::Target: Logger
+	L::Target: Logger,
 {
 	fn write_to_file(&self, writer: &mut fs::File) -> Result<(), std::io::Error> {
@@ -96,5 +96,5 @@ impl FilesystemPersister {
 		K::Target: KeysInterface<Signer=Signer>,
 		F::Target: FeeEstimator,
-		L::Target: Logger
+		L::Target: Logger,
 	{
 		let path = PathBuf::from(data_dir);
@@ -106,5 +106,5 @@ impl FilesystemPersister {
 		&self, keys_manager: K
 	) -> Result<Vec<(BlockHash, ChannelMonitor<Signer>)>, std::io::Error>
-		where K::Target: KeysInterface<Signer=Signer> + Sized
+		where K::Target: KeysInterface<Signer=Signer> + Sized,
 	{
 		let path = self.path_to_monitor_data();
$
```

**Pull request merged**